### PR TITLE
fix static analysis complaints

### PR DESCRIPTION
**Pull request by rileyberton**
clang scan-build uncovered these obvious things

### CODE REVIEW

File: src/eventer/eventer_SSL_fd_opset.c [hunk 1]
```diff
@@ -626,7 +626,6 @@ eventer_ssl_ctx_new(eventer_ssl_orientation_t type,
   layer_str = layer ? layer : DEFAULT_LAYER_STRING;
   ctx_layer = alloca(strlen(layer_str)+1);
   memcpy(ctx_layer, layer_str, strlen(layer_str)+1);
-  layer_str = ctx_layer;
```
Comment (rileyberton) [inline src/eventer/eventer_SSL_fd_opset.c]: Never used after this

File: src/eventer/eventer_jobq.c [hunk 2]
```diff
@@ -315,7 +315,7 @@ eventer_jobq_execute_timeout(eventer_t e, int mask, void *closure,
       memcpy(jobcopy, job, sizeof(*jobcopy));
       free(job);
       jobcopy->fd_event = my_precious;
-      job->finish_hrtime = eventer_gethrtime();
+      jobcopy->finish_hrtime = eventer_gethrtime();
```
Comment (rileyberton) [inline src/eventer/eventer_jobq.c]: use after free

File: src/mtev_conf.c [hunk 3]
```diff
@@ -537,7 +537,7 @@ remove_emancipated_child_node(xmlNodePtr oldp, xmlNodePtr node) {
     if (node->next) node->next->prev = node->prev;
   }
   else {
-    xmlNodePtr prev = oldp->children;
+    xmlNodePtr prev;
```
Comment (rileyberton) [inline src/mtev_conf.c]: double init

File: src/mtev_console.c [hunk 4]
```diff
@@ -426,7 +426,7 @@ mtev_console_std(eventer_t e, int mask, void *closure,
   }
 
   if(mtev_console_continue_sending(ncct, &newmask) < 0) {
-    if(ncct->wants_shutdown || errno != EAGAIN) goto socket_error;
+    if((ncct != NULL && ncct->wants_shutdown) || errno != EAGAIN) goto socket_error;
```
Comment (rileyberton) [inline src/mtev_console.c]: missing NULL check

File: src/mtev_net_heartbeat.c [hunk 5]
```diff
@@ -244,7 +244,6 @@ mtev_net_heartbeat_serialize_and_send(mtev_net_heartbeat_ctx *ctx) {
       mtevL(mtev_error, "netheartbeat: malloc(%d) failure\n", len + blocksize * 2);
       goto bail;
     }
-    cipher_buf_len = len + blocksize * 2;
```
Comment (rileyberton) [inline src/mtev_net_heartbeat.c]: never used

File: src/mtev_reverse_socket.c [hunk 6]
```diff
@@ -911,7 +911,6 @@ int mtev_reverse_socket_connect(const char *id, int existing_fd) {
   pthread_rwlock_rdlock(&reverse_sockets_lock);
   if(mtev_hash_retrieve(&reverse_sockets, id, strlen(id), &vrc)) {
     rc = vrc;
-    i = rc->data.last_allocated_channel + 1;
```
Comment (rileyberton) [inline src/mtev_reverse_socket.c]: dupe init

File: src/mtev_reverse_socket.c [hunk 7]
```diff
@@ -1611,7 +1610,6 @@ mtev_reverse_client_handler(eventer_t e, int mask, void *closure,
     family = family == AF_INET ? AF_INET6 : AF_INET;
     rv = inet_pton(family, target, &a);
     if(rv != 1) {
-      family = AF_INET;
```
Comment (rileyberton) [inline src/mtev_reverse_socket.c]: never used

File: src/utils/mtev_log.c [hunk 8]
```diff
@@ -701,7 +701,6 @@ posix_logio_size(mtev_log_stream_t ls) {
   asynch_log_ctx *actx = ls->op_ctx;
   pthread_rwlock_t *lock = ls->lock;
   if(lock) pthread_rwlock_rdlock(lock);
-  actx = ls->op_ctx;
```
Comment (rileyberton) [inline src/utils/mtev_log.c]: dupe init

File: src/utils/mtev_skiplist.c [hunk 9]
```diff
@@ -94,7 +94,7 @@ void mtev_skiplist_set_compare(mtev_skiplist *sl,
 void mtev_skiplist_add_index(mtev_skiplist *sl,
                              mtev_skiplist_comparator_t comp,
                              mtev_skiplist_comparator_t compk) {
-  mtev_skiplist_node *m;
+  mtev_skiplist_node *m = NULL;
```
Comment (rileyberton) [inline src/utils/mtev_skiplist.c]: missing init

**Comment by pamaddox** [comment]
This looks fine to me.

**Pull request merged**